### PR TITLE
Iss864

### PR DESCRIPTION
**Pull request by normangraf**
I removed superfluous code in ReconParticleDriver that accessed the rotated helical track hits for no reason.
One can now run the Kalman Filter track finding and fitting without having to create HelicalTrackHits.
The [KF only steering file](https://github.com/JeffersonLab/hps-java/compare/master...iss864#diff-486de823c08371045aebe80b2e41e45995f8031b417d0fbfa0b1f97bdf615272) has been modified to remove that Driver.
It has also been updated to use the distance-based track cluster matcher.
It also now keeps monster events instead of zeroing out the 1D strip hit collection.

### CODE REVIEW

**Comment by cbravo135** [comment]
Can you show some checks that this did not change the behavior of PhysicsRun2019_pass0_recon_evio.lcsim?

**Comment by normangraf** [comment]
The two collections that were being accessed were never being used. Commenting out those calls should not affect the results of any other Drivers.

File: steering-files/src/main/resources/org/hps/steering/recon/PhysicsRun2019_pass0_KFOnly.lcsim [hunk 1]
```diff
@@ -87,7 +89,7 @@
        
         <driver name="TrackerHitDriver" type="org.hps.recon.tracking.DataTrackerHitDriver">
             <neighborDeltaT>8.0</neighborDeltaT>
-            <saveMonsterEvents>false</saveMonsterEvents>
+            <saveMonsterEvents>true</saveMonsterEvents>
```
Comment (cbravo135) [inline steering-files/src/main/resources/org/hps/steering/recon/PhysicsRun2019_pass0_KFOnly.lcsim:92]: Why is this changed from false to true?

**Comment by cbravo135** [comment]
I think at a minimum you should run on one file and show that these collections are indeed removed as expected.

**Comment by cbravo135** [comment]
Also, if we plan on moving forward with this steering file now, you should add the EventFlagFilter driver and turn on the "svt_readout_overlap_good" flag.

**Comment by normangraf** [comment]
Seems to be a "Fitting with: Simplex" debug printout left in the code.


Ooops! Added this comment to the wrong PR.

File: steering-files/src/main/resources/org/hps/steering/recon/PhysicsRun2019_pass0_KFOnly.lcsim (hunk 1 above)
Comment (normangraf) [reply steering-files/src/main/resources/org/hps/steering/recon/PhysicsRun2019_pass0_KFOnly.lcsim:92]: I don't believe the current way of handling these events is correct, and simply skipping these events may be giving us a false impression of how well we're doing. I'd like to see what effect this has in combination with enabling the event flag filter.

**Comment by normangraf** [comment]
The collections aren't (yet) being removed. I was simply removing an extraneous call to access those collections, which requires them to be in the event. Once we drop the SeedTracker  and GBL track finding and fitting these collections will no longer be in the event. No behavior has changed.

**Review by bloodyyugo — APPROVED** [review]

File: steering-files/src/main/resources/org/hps/steering/recon/PhysicsRun2019_pass0_KFOnly.lcsim (hunk 1 above)
Comment (cbravo135) [reply steering-files/src/main/resources/org/hps/steering/recon/PhysicsRun2019_pass0_KFOnly.lcsim:92]: The event flag filter is not currently activated in this steering file.

Comment (cbravo135) [reply steering-files/src/main/resources/org/hps/steering/recon/PhysicsRun2019_pass0_KFOnly.lcsim:92]: If this is your intention, please add it to this steering file and I will approve this PR.

**Review by pbutti — APPROVED** [review]
